### PR TITLE
feat: extract static finalizer validation method

### LINKED ISSUE #3734: Extract finalizer validation so that it can be used statically instead of only on a `HasMetadata` instance
### Is your enhancement related to a problem? Please describe

Right now, you need to create an `HasMetadata` instance to validate a finalizer. The logic should be available without having to do so.

### Describe the solution you'd like

Extract the validation method into a static method.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by metacosm**
Fixes #3734

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [x] I Added [CHANGELOG](../CHANGELOG.md) entry regarding this change
 - [x] I have implemented unit tests to cover my changes
 - [x] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

File: kubernetes-model-generator/kubernetes-model-core/src/main/java/io/fabric8/kubernetes/api/model/HasMetadata.java [hunk 1]
```diff
@@ -228,12 +228,13 @@ default boolean addFinalizer(String finalizer) {
 
   /**
    * Determines whether the specified finalizer is valid according to the
-   * <a href='https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#finalizer'>finalizer specification</a>.
+   * <a href='https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#finalizer'>finalizer
```
Comment (metacosm) [inline kubernetes-model-generator/kubernetes-model-core/src/main/java/io/fabric8/kubernetes/api/model/HasMetadata.java:231]:
```suggestion
   * <a href='https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#finalizers'>finalizer
```

**Review by iocanel — APPROVED** [review]
LGMT

**Review by shawkins — APPROVED** [review]
LGTM

@metacosm do you want me to capture an issue about a general approach to client side validation?

**Comment by metacosm** [comment]
> LGTM
> 
> @metacosm do you want me to capture an issue about a general approach to client side validation?

Would be interesting, indeed!

**Review by sunix — COMMENTED** [review]
I suggest having the final commit message like

> refactor: extract static finalizer validation method 

or

> feat: Allow finalizer validation method to be used statically instead of only on a HasMetadata instance

**Comment by metacosm** [comment]
> I suggest having the final commit message like
> 
> > refactor: extract static finalizer validation method

Won't the commits be squashed anyway?

**Comment by sunix** [comment]
> Won't the commits be squashed anyway?

It depends who is "merging" it

**Review by manusa — APPROVED** [review]

**Pull request merged**